modules/tensor_mechanics/include/utils/RankTwoScalarTools.h

```diff
@@ -467,4 +472,86 @@ getQuantity(const RankTwoTensorTempl<T> & tensor,
                  scalarOptions().getRawNames());
   }
 }
+
+template <typename T>
+T
+getCylindricalComponent(const RankTwoTensorTempl<T> & tensor,
+                        const MooseEnum & scalar_type,
+                        const Point & point1,
+                        const Point & point2,
+                        const Point & curr_point,
+                        Point & direction)
+{
+
+  if (scalar_type == "AxialStress")
+    return axialStress(tensor, point1, point2, direction);
+  else if (scalar_type == "HoopStress")
+    return hoopStress(tensor, point1, point2, curr_point, direction);
+  else if (scalar_type == "RadialStress")
+    return radialStress(tensor, point1, point2, curr_point, direction);
+  else
+    mooseError("RankTwoCylindricalComponent Error: Pass valid scalar type - " +
+               cylindricalOptions().getRawNames());
+}
+
+template <typename T>
+T
+getPrincipalComponent(const RankTwoTensorTempl<T> & tensor,
+                      const MooseEnum & scalar_type,
+                      Point & direction)
+{
+  if (scalar_type == "MaxPrincipal")
+    return maxPrincipal(tensor, direction);
+  else if (scalar_type == "MidPrincipal")
+    return midPrincipal(tensor, direction);
+  else if (scalar_type == "MinPrincipal")
+    return minPrincipal(tensor, direction);
+  else
+    mooseError("RankTwoInvariant Error: Pass valid invariant - " +
+               principalComponentOptions().getRawNames());
+}
+
+template <typename T>
+T
+getDirectionalComponent(const RankTwoTensorTempl<T> & tensor,
+                        const MooseEnum & scalar_type,
+                        const Point & direction)
+{
+  if (scalar_type == "Direction")
+    return directionValueTensor(tensor, direction);
+  else
+    mooseError("RankTwoDirectionalComponent Error: Pass valid invariant - " +
+               directionOption().getRawNames());
+}
+
+template <typename T>
+T
+getCartesianComponent(const RankTwoTensorTempl<T> & tensor, const MooseEnum & scalar_type)
```

Comment:
This should be something like `getInvariant`.  These aren't Cartesian components.